tests/unittests/config/test_apt_source_v1.py

```diff
@@ -44,51 +46,46 @@ def update_package_sources(self):
         return
 
 
-class TestAptSourceConfig(TestCase):
+class TestAptSourceConfig:
     """TestAptSourceConfig
     Main Class to test apt_source configs
     """
 
     release = "fantastic"
-
-    def setUp(self):
-        super(TestAptSourceConfig, self).setUp()
-        self.tmp = tempfile.mkdtemp()
-        self.addCleanup(shutil.rmtree, self.tmp)
-        self.aptlistfile = os.path.join(self.tmp, "single-deb.list")
-        self.aptlistfile2 = os.path.join(self.tmp, "single-deb2.list")
-        self.aptlistfile3 = os.path.join(self.tmp, "single-deb3.list")
-        self.join = os.path.join
-        self.matcher = re.compile(ADD_APT_REPO_MATCH).search
-        # mock fallback filename into writable tmp dir
-        self.fallbackfn = os.path.join(
-            self.tmp, "etc/apt/sources.list.d/", "cloud_config_sources.list"
+    matcher = re.compile(ADD_APT_REPO_MATCH).search
+
+    @pytest.fixture
+    def apt_lists(self, tmpdir):
```

Comment:
I like that you created a separate fixture here for use only in tests that care about those paths.